x-pack/packages/security-solution/navigation/src/links.tsx

```diff
@@ -0,0 +1,142 @@
+
+export interface WrappedLinkProps {
+  id: string;
+  path?: string;
+  urlState?: string;
+}
+
+export interface LinkProps {
+  onClick: MouseEventHandler;
+  href: string;
+}
+
+export type GetLinkUrl = (
+  params: WrappedLinkProps & {
+    absolute?: boolean;
+    urlState?: string;
+  }
+) => string;
+
+export type GetLinkProps = (
+  params: WrappedLinkProps & {
+    /**
+     * Optional `onClick` callback prop.
+     * It is composed within the returned `onClick` function to perform extra actions when the link is clicked.
+     * It does not override the navigation operation.
+     **/
+    onClick?: MouseEventHandler;
+  }
+) => LinkProps;
+
+/**
+ * It returns the `url` to use in link `href`.
+ */
+export const useGetLinkUrl = () => {
+  const { getAppUrl } = useGetAppUrl();
+
+  const getLinkUrl = useCallback<GetLinkUrl>(
+    ({ id, path = '', absolute = false, urlState }) => {
+      const formattedPath = urlState ? formatPath(path, urlState) : path;
+      const { appId, deepLinkId } = getAppIdsFromId(id);
+      return getAppUrl({ deepLinkId, appId, path: formattedPath, absolute });
+    },
+    [getAppUrl]
+  );
+
+  return getLinkUrl;
+};
+
+/**
+ * It returns the `onClick` and `href` props to use in link components based on the` deepLinkId` and `path` parameters.
+ */
+export const useGetLinkProps = (): GetLinkProps => {
+  const getLinkUrl = useGetLinkUrl();
+  const { navigateTo } = useNavigateTo();
+
+  const getLinkProps = useCallback<GetLinkProps>(
+    ({ id, path, urlState, onClick: onClickProps }) => {
+      const url = getLinkUrl({ id, path, urlState });
+      return {
+        href: url,
+        onClick: (ev: MouseEvent) => {
+          if (isModified(ev)) {
+            return;
+          }
+          if (onClickProps) {
+            onClickProps(ev);
+          }
+          ev.preventDefault();
+          navigateTo({ url });
+        },
+      };
+    },
+    [getLinkUrl, navigateTo]
+  );
+
+  return getLinkProps;
+};
+
+/**
+ * HOC that wraps any Link component and makes it a navigation Link.
+ */
+export const withLink = <T extends Partial<LinkProps>>(
+  Component: React.ComponentType<T>
+): React.FC<Omit<T & WrappedLinkProps, 'href'>> =>
+  React.memo(function ({ id, path, urlState, onClick: _onClick, ...rest }) {
+    const getLink = useGetLinkProps();
+    const { onClick, href } = getLink({ id, path, urlState, onClick: _onClick });
+    return <Component onClick={onClick} href={href} {...(rest as unknown as T)} />;
+  });
+
+/**
+ * Security Solutions internal link button.
+ *
+ * `<LinkButton deepLinkId={SecurityPageName.hosts} />;`
+ */
+export const LinkButton = withLink(EuiButton);
+
+/**
+ * Security Solutions internal link anchor.
+ *
+ * `<LinkAnchor deepLinkId={SecurityPageName.hosts} />;`
+ */
+export const LinkAnchor = withLink<EuiLinkProps>(EuiLink);
+
+// Utils
+
+export const isExternalId = (id: string): boolean => id.includes(':');
```

Comment:
We need to add here a comment/example, which will explain why we are using search of ':' in id for marking it as an external.